### DISCUSSION TITLE
refresh async atom without Suspense loading

### DISCUSSION THREAD

**Discussion by lili21** [pmndrs/jotai#1102] [Q&A] [answered]
I need the suspense loading at the initial loading. 
but don't need it when refresh async atom . 

any suggestion?

**Comment by dai-shi** [comment, marked as answer]
`atomWithDefault` should do.

**Reply by lili21** [reply]
https://codesandbox.io/s/interesting-wildflower-j2vvnq?file=/src/App.js

it does't work as expect.

**Reply by dai-shi** [reply]
After initial loading, you want to set resolved value because you want to avoid suspense.
The refetchAtom pattern (with versionAtom) is a kind of workaround to force suspense for refetch.

something like:
```js
const anAtom = atomWithDefault(() => {
  // initial loading
})
const refetchAtom = atom(null, async (get, set, arg) => {
  // refetch
  set(anAtom, resolvedValue)
})
```

**Comment by Preston-Harrison** [comment]
Could you explain this? I'm pretty lost here.

**Reply by Preston-Harrison** [reply]
Ended up with this as a solution for an atom that started with an initial async value (that triggered a suspense), followed by refreshes that updated the value but did not trigger suspense. Please let me know if there is a bug in my implementation, or if there is a better way to do this.
```typescript
import { atom, Getter } from "jotai";
import { atomWithDefault } from "jotai/utils";

export function atomWithRefresh<T>(fn: (get: Getter) => T) {
	const initial = atomWithDefault(fn);
	const thisAtom = atom(get => get(initial), async (get, set) => {
		const next = await fn(get);
		set(initial, next);
	});
	return thisAtom;
}
```

**Reply by dai-shi** [reply]
This is interesting. It should work. I don't see anything wrong in the implementation. It could be implemented without atomWithDefault if you are interested.
